misk/src/main/kotlin/misk/hibernate/ReflectionQueryFactory.kt

```diff
@@ -32,16 +38,24 @@ internal class ReflectionQuery<T : DbEntity<T>>(
   private val rootEntityType: KClass<T>,
   private val queryMethodHandlers: Map<Method, QueryMethodHandler>
 ) : Query<T>, InvocationHandler {
+  override var maxRows = -1
```

Comment:
why the default of -1 instead of just MAX_MAX_ROW?

Reply:
If the user provides a limit, we happily truncate. If we add the limit ourselves and need to truncate, that’s an error.

Reply:
lemme add one more test for that.